File: ts/example/sort/inheritance/Dog.ts
import { String } from '@otterop/lang/String';
import { Animal } from './Animal';

export class Dog extends Animal {

    private _dogWord : String;

    public constructor() {
        super(String.wrap("I don't know"));
        this._dogWord = String.wrap("bark");
    }

    public act() : String {
        super.act();
        return this._dogWord;
    }
}

File: dotnet/src/Example/Sort/Inheritance/Dog.cs
namespace Example.Sort.Inheritance
{
    public class Dog : Animal
    {
        private Otterop.Lang.String dogWord;
        public Dog() : base(Otterop.Lang.String.Wrap("I don't know"))
        {
            this.dogWord = Otterop.Lang.String.Wrap("bark");
        }

        public new Otterop.Lang.String Act()
        {
            base.Act();
            return this.dogWord;
        }

    }

}
